auctioneer/client.go

```diff
@@ -968,9 +989,13 @@ func (c *Client) readIncomingStream() { // nolint:gocyclo
 				return
 			}
 
-			// Any other error we want to report back.
+			log.Errorf("Server connection error received: %v", err)
+
+			// For any other error type, we'll attempt to trigger
+			// the reconnect logic so we'll always try to connect
+			// to the server in the background.
 			select {
-			case c.errChanSwitch.ErrChan() <- err:
+			case c.errChanSwitch.ErrChan() <- ErrServerShutdown:
```

Comment:
This was from the original commit, right? Do we still want this? 

IIRC, the error assertion elsewhere in the codebase would cause us to actually _not_ reconnect (if this error is sent) since we assumed that a reconnection was already in place.

Reply:
Ah, you're right. Created a specific error for this.